### PR TITLE
Add internal solution for Air Cownditioning II (usaco-1276)

### PR DESCRIPTION
**Pull request by ChuangSheep**
WIP

---
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/bronze/usaco-1276.mdx [hunk 1]
```diff
@@ -143,7 +143,7 @@ using namespace std;
  * Check if the given 'air_conditioners' satisfy the constraints of the cows
  */
 bool check(vector<array<int, 3>> &cows,
-		   vector<array<int, 4>> &air_conditioners) {
+           vector<array<int, 4>> &air_conditioners) {
```
Comment (ChuangSheep) [inline solutions/bronze/usaco-1276.mdx:146]: c'mon the ci removes my tabs and converts them to spaces??

File: solutions/bronze/usaco-1276.mdx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int N, M;
+// {s, t, c}
+vector<array<int, 3>> cows;
+// {a, b, p, m}
+vector<array<int, 4>> air_conditioners;
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1276.mdx:40]: use structs

File: solutions/bronze/usaco-1276.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/bronze/usaco-1276.mdx:146]: tabs for indentation, spaces for formatting
stupid, but hey, what u gon do about it

File: solutions/bronze/usaco-1276.mdx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+	}
+
+	vector<array<int, 4>> air_conditioners;
+	for (int i = 0; i < M; i++) {
+		int a, b, p, m;
+		cin >> a >> b >> p >> m;
+		air_conditioners.push_back({a, b, p, m});
+	}
+
+	int min_cost = 1e9;
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1276.mdx:190]: int32_max here asw

File: solutions/bronze/usaco-1276.mdx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+
+## Solution 2: Subset with Bitmask
+
+**Time Complexity:** $\mathcal{O}(2^M \cdot 100 \cdot (M + N))$
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1276.mdx:133]: might i suggest putting [my other language solutions](https://bitbucket.org/SansPapyrus683/workspace/snippets/nE4AnK/usaco-2023-january-bronze-air) here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/bronze/usaco-1276.mdx (hunk 4 above)
Comment (ChuangSheep) [reply solutions/bronze/usaco-1276.mdx:133]: sure

File: solutions/bronze/usaco-1276.mdx (hunk 2 above)
Comment (ChuangSheep) [reply solutions/bronze/usaco-1276.mdx:40]: I seriously do not think struct provides any clearity here: we are using auto structured binding already. Declaring two extra structs just makes the code unnecessarily longer

Comment (SansPapyrus683) [reply solutions/bronze/usaco-1276.mdx:40]: if you insits

File: solutions/bronze/usaco-1276.mdx [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+satisfy the requirements of the cows by iterating through all possible positions
+$i \in [1;100]$. For each position $i$, we can iterate through all available air
+conditioners in the current subset to find out the temperature reduced at $i$
+and iterate through all cows to find the cow at this position and its demand. If
+the requirement at each position is fulfilled, we update the minimal cost
+accordingly.
+
+There are two ways to generate all subsets of air conditioners: Using recursion
+or a bitmask.
+
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1276.mdx:23]:
```suggestion
We can generate subsets with recursion or bitmasks: both ways are presented below.
```

File: solutions/bronze/usaco-1276.mdx [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+		public int start, end;
+		public int coolAmt;
+		public int cost;
+		public AC(int start, int end, int coolReq, int cost) {
+			this.start = start;
+			this.end = end;
+			this.coolAmt = coolReq;
+			this.cost = cost;
+		}
+	}
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1276.mdx:238]:
```suggestion
	//BeginCodeSnip{Cow and AC Class}
	static class Cow {
		public int start, end;
		public int coolReq;
		public Cow(int start, int end, int coolReq) {
			this.start = start;
			this.end = end;
			this.coolReq = coolReq;
		}
	}
	static class AC {
		public int start, end;
		public int coolAmt;
		public int cost;
		public AC(int start, int end, int coolReq, int cost) {
			this.start = start;
			this.end = end;
			this.coolAmt = coolReq;
			this.cost = cost;
		}
	}
	//EndCodeSnip
```

**Review by SansPapyrus683 — APPROVED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
@envyaims @ryanchou-dev

**Review by ryanchou-dev — APPROVED** [review]
looks good!

**Pull request merged**